pkg/operation/seed/scheduler/gardener_kube_scheduler.go

```diff
@@ -0,0 +1,367 @@
+											Name: configMap.Name,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		serviceAccount = &corev1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: k.namespace,
+			Labels:    getLabels(),
+		}}
+		roleBinding = &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      roleBindingName,
+				Namespace: metav1.NamespaceSystem,
+				Labels:    getLabels(),
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "Role",
+				Name:     roleName,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      name,
+				Namespace: k.namespace,
+			}},
+		}
+		webhook = &admissionregistrationv1beta1.MutatingWebhookConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   webhookName,
+				Labels: getLabels(),
+			},
+			Webhooks: []admissionregistrationv1beta1.MutatingWebhook{{
+				Name:         webhookName,
+				ClientConfig: *k.webhookClientConfig,
+				Rules: []admissionregistrationv1beta1.RuleWithOperations{{
+					Operations: []admissionregistrationv1beta1.OperationType{admissionregistrationv1beta1.Create},
+					Rule: admissionregistrationv1beta1.Rule{
+						APIGroups:   []string{corev1.GroupName},
+						APIVersions: []string{corev1.SchemeGroupVersion.Version},
+						Scope:       &scope,
+						Resources:   []string{"pods"},
+					},
+				}},
+				FailurePolicy: &failPolicy,
+				MatchPolicy:   &matchPolicy,
+				NamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						v1beta1constants.GardenRole: v1beta1constants.GardenRoleShoot,
+					},
+				},
+				ObjectSelector:          &metav1.LabelSelector{},
+				SideEffects:             &sideEffects,
+				TimeoutSeconds:          &timeout,
+				AdmissionReviewVersions: []string{admissionregistrationv1beta1.SchemeGroupVersion.Version},
+				ReinvocationPolicy:      &revocationPolicy,
+			}},
+		}
+		registry = managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
+	)
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.client, namespace, func() error {
+		namespace.Labels = getLabels()
+		return nil
+	}); err != nil {
+		return errors.Wrap(err, "update of Namespace failed")
+	}
+
+	resources, err := registry.AddAllAndSerialize(
+		kubeSchedulerClusterRoleBinding,
+		volumeSchedulerClusterRoleBinding,
+		roleBinding,
+		serviceAccount,
+		configMap,
+		deployment,
+		webhook,
+	)
+	if err != nil {
+		return err
+	}
+
+	return common.DeployManagedResourceForSeed(ctx, k.client, "gardener-kube-scheduler", k.namespace, false, resources)
+}
+
+func getLabels() map[string]string {
+	return map[string]string{
+		"app":  "kubernetes",
+		"role": "scheduler",
+	}
+}
+
+func (k *kubeScheduler) Destroy(ctx context.Context) error {
+	return common.DeleteManagedResourceForSeed(ctx, k.client, "gardener-kube-scheduler", k.namespace)
```

Comment:
You'd have to delete the namespace here if you created it manually yourself, right?